cli/pkg/providers/provider.go

```diff
@@ -21,6 +21,7 @@ const (
 type Provider interface {
 	Provision(providerConfig map[string]string, dir workspace.ConfigDir) (kube.KubeCtx, error)
 	GetOverrides() template.Variables
+	IsEnvironmentValid() (bool, error)
```

Comment:
Perhaps it is better to return just `error`. As if `err := nil`, means that the check failed.